### PR TITLE
fix(git): ignore symbolic links for getFileList

### PR DESCRIPTION
**Pull request by rarkins**

### CODE REVIEW

**Comment by JamieMagee** [comment]
The `substring(53)` doesn't work for submodules. They have type `commit` instead of `blob` so it needs to be `substring(55)`.

It might be easier to split on `\s`?

File: lib/platform/git/storage.ts [hunk 1]
```diff
@@ -308,19 +308,16 @@ export class Storage {
...
+    const files: string = await this._git.raw(['ls-tree', '-r', branch]);
     // istanbul ignore if
     if (!files) {
       return [];
     }
     return files
       .split('\n')
       .filter(Boolean)
+      .filter((line) => !line.startsWith('120000'))
+      .map((line) => line.split(/\s/).pop())
```
Comment (rarkins) [inline lib/platform/git/storage.ts:320]: @JamieMagee looks safe now?

Comment (viceice) [inline lib/platform/git/storage.ts:320]:
```suggestion
      .filter((line) => line.startsWith('100'))
      .map((line) => line.split(/\t/).pop())
```
https://git-scm.com/docs/git-ls-tree#_output_format
https://stackoverflow.com/a/737877/10109857

**Review by viceice — CHANGES_REQUESTED** [review]

File: lib/platform/git/storage.ts (hunk 1 above)
Comment (viceice) [reply lib/platform/git/storage.ts:320]: @rarkins We only need regular files, which starts with `100`

Comment (JamieMagee) [reply lib/platform/git/storage.ts:320]: See @viceice comment. I didn't realise it's `\t` instead of `\s`.

**Pull request merged**